homeassistant/components/speedtestdotnet/config_flow.py

```diff
@@ -0,0 +1,88 @@
+"""Config flow for Transmission Bittorent Client."""
+import speedtest
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_SCAN_INTERVAL
+from homeassistant.core import callback
+
+from .const import (
+    CONF_MANUAL,
+    CONF_SERVER_ID,
+    DEFAULT_NAME,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+)
+
+
+class SpeedTestFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle SpeedTest config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return SpeedTestOptionsFlowHandler(config_entry)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        if self._async_current_entries():
+            return self.async_abort(reason="one_instance_allowed")
+
+        if user_input is None:
+            return self.async_show_form(step_id="user")
+
+        return self.async_create_entry(title=DEFAULT_NAME, data=user_input)
+
+    async def async_step_import(self, import_config):
+        """Import from SpeedTest config."""
+        import_config[CONF_SCAN_INTERVAL] = (
+            import_config[CONF_SCAN_INTERVAL].seconds / 60
```

Comment:
As this is an internet speed test my idea was to make the minimum update interval 1 min. because if the user defines the update interval as 1 second (for example) then the speed test will run each 1 second and this is not good as this integration requires downloading and uploading to get the results.
We can mention in the documentation that minimum interval is 1 minute.